cnf-certification-test/accesscontrol/suite.go

```diff
@@ -209,8 +209,30 @@ var _ = ginkgo.Describe(common.AccessControlTestKey, func() {
 		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
 		test1337UIDs(&env)
 	})
+
+	testID, tags = identifiers.GetGinkgoTestIDAndLabels(identifiers.TestAutomountToken)
+	ginkgo.It(testID, ginkgo.Label(tags...), func() {
+		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
+		testAutomountToken(&env)
+	})
 })
 
+func testAutomountToken(env *provider.TestEnvironment) {
+	ginkgo.By("Testing pods to ensure using projected volume")
+	var badPods []string
+	for _, put := range env.Pods {
+		for i := range put.Spec.Volumes {
+			if put.Spec.Volumes[i].Projected.Sources == nil {
```

Comment:
```suggestion
			if put.Spec.Volumes[i].Projected.Sources != nil {
```

I think this needs to change to see if `!= nil` because we want to check the pod spec to see if there is a projected volume.  And if there is a projected volume, we want to go one step further and check the `sources` for `serviceAccountToken`:
```
sources:
      - serviceAccountToken:
          audience: api
          expirationSeconds: 3600
          path: token
```

Reply:
If `sources` == nil, then `continue`.

Reply:
**if sources == nil** , it "badpods" and I will append to list "badpods".
**if sources!=nil**  , I will check if ServiceAccountToken exist it is good because we have serviceAccount and I will print details about the pod. tell me if we need more details for pod (when I printed)